### PR TITLE
feat: support external registries

### PR DESCRIPTION
**Pull request by johanneswuerbach**
Support pushing to external registries instead of the Humanitec provided one.

Example usage would be:

```
- name: Build and Push to Humanitec
  uses: humanitec/build-push-to-humanitec@v1
  with:
    external-registry: ghcr-package

### CODE REVIEW

**Comment by F3licity** [comment]
> Example usage would be:
> 
> ```
> - name: Build and Push to Humanitec
>   uses: humanitec/build-push-to-humanitec@v1
>   with:
>     external-registry: ghcr-package
> ```

since we are pushing to an external registry, called `ghcr-package`, shouldn't it be like this:..?
`-name: Build and Push to ghcr-package` ? 🤔

**Comment by Nilsty** [comment]
what about an option to provide username and password for an external registry?

**Comment by johanneswuerbach** [comment]
> what about an option to provide username and password for an external registry?

Not sure this is needed, if people need to authenticate they can do so before calling this action and authentication can be fairly different on ghcr/gcp/aws.

E.g. in our case we use workload identity so an explicit authentication isn't required.

**Comment by F3licity** [comment]
> In any case the `name` property is any string the user wants and this was just an example :-)

yes ofc, I was just trying to confirm whether I understood the aim of your PR properly.

File: action.yaml [hunk 1]
```diff
@@ -38,6 +38,10 @@ inputs:
     description: 'Allows the default humanitec api to be overidden for testing.'
     required: false
     default: 'api.humanitec.io'
+  external-registry:
```
[the remark this answers is not in the document]
Comment (johanneswuerbach) [reply action.yaml:41]: Not sure I follow. This parameter only allows to configure which registry is targeted by docker push and when registering the artefact.

This registry can be registered or not. In our case we don't need the register registry as all cluster instances pull our images using workload identity anyway.

[the remark this answers is not in the document]
Comment (johanneswuerbach) [reply action.yaml]: Updated the document, let me know if this makes it more clear.

**Review by Nilsty — APPROVED** [review]

**Pull request merged**